modules/library/coverage/src/test/java/org/geotools/image/ImageWorkerTest.java

```diff
@@ -1713,6 +1720,40 @@ public void testBandMerge() {
         assertNoData(image, null);
     }
 
+    @Test
+    public void testBandMergeWithNodata() throws FileNotFoundException, IOException {
+        double noDataValue = -10000d;
+        Range nodata = RangeFactory.create(noDataValue, noDataValue);
+        double[] background = new double[] {noDataValue};
+
+        RenderedImage twoBands = null;
+        PlanarImage pi = PlanarImage.wrapRenderedImage(imageWithNodata2);
+        pi.setProperty("GC_NODATA", new NoDataContainer(nodata));
+
+        ImageWorker worker =
+                new ImageWorker()
```

Comment:
So why the mosaic in the test? The code had no changes there or not?

Reply:
Just to add something in the mix :) This was one of the chains initially spotting the issue.
Going to check if the problem still happen after a simple read.